robotiq_s_model_visualization/cfg/robotiq_hand.urdf.xacro

```diff
@@ -0,0 +1,798 @@
+
+      <!-- Link palm -->
+      <link name="${prefix}palm">
+        <visual>
+          <geometry>
+            <mesh filename="package://robotiq_s_model_visualization/meshes/s-model_articulated/visual/palm.STL" />
+          </geometry>
+          <material name="green">
+            <color rgba="0 1 0 1"/>
+          </material>
+        </visual>
+        <!-- Simple shape collision -->
+        <collision>
+          <origin rpy="0 0 0" xyz="0 0 0"/>
+          <geometry>
+            <box size="0.1 0.1 0.1"/>
+          </geometry>
+          <material name="yellow">
+            <color rgba="0 1 1 1"/>
+          </material>
+        </collision>
+        <inertial>
+          <origin rpy="0 0 0" xyz="0 0 0"/>
+          <mass value="1.3"/>
+          <inertia ixx="0.0035" ixy="0" ixz="-0" iyy="0.0035" iyz="0" izz="0.0035"/>
+        </inertial>
+      </link>
+
+      <gazebo reference="${prefix}palm">
+        <mu1>1.1</mu1>
+        <mu2>1.1</mu2>
+      </gazebo>
+    <!-- End of link list -->
+
+    <!-- Joint list -->
+
+    <!-- Joint palm_finger_1_joint -->
+    <joint name="${prefix}palm_finger_1_joint" type="revolute">
+      <parent link="${prefix}palm"/>
+      <child link="${prefix}finger_1_link_0"/>
+      <dynamics damping="1" friction="0" />
+      <axis xyz="0 1 0"/>
+      <origin rpy="0 3.1416 -1.57" xyz="-0.0455 0.0214 0.036"/>
+      <!-- Limits: [-17deg.,17deg.] -->
+      <limit lower="-0.2967" upper="0.2967" effort="60" velocity="-1"/>
+    </joint>
+
+    <gazebo reference="${prefix}palm_finger_1_joint">
+      <implicitSpringDamper>1</implicitSpringDamper>
+    </gazebo>
+
+    <!-- Joint palm_finger_2_joint -->
+    <joint name="${prefix}palm_finger_2_joint" type="revolute">
+      <parent link="${prefix}palm"/>
+      <child link="${prefix}finger_2_link_0"/>
+      <dynamics damping="1" friction="0" />
+      <axis xyz="0 1 0"/>
+      <origin rpy="0 3.1416 -1.57" xyz="-0.0455 0.0214 -0.036"/>
+      <!-- Limits: [-17deg.,17deg.] -->
+      <limit lower="-0.2967" upper="0.2967" effort="60" velocity="-1"/>
+    </joint>
+
+    <gazebo reference="${prefix}palm_finger_2_joint">
+      <implicitSpringDamper>1</implicitSpringDamper>
+    </gazebo>
+
+    <!-- Joint palm_finger_middle_joint -->
+    <joint name="${prefix}palm_finger_middle_joint" type="revolute">
+      <parent link="${prefix}palm"/>
+      <child link="${prefix}finger_middle_link_0"/>
+      <dynamics damping="1" friction="0" />
+      <axis xyz="0 1 0"/>
+      <origin rpy="0 0 1.57" xyz="0.0455 0.0214 0"/>
+      <limit lower="0" upper="0" effort="60" velocity="-1"/>
+    </joint>
+
+    <gazebo reference="${prefix}palm_finger_middle_joint">
+      <implicitSpringDamper>1</implicitSpringDamper>
+    </gazebo>
+    <!-- end of joint list -->
+
+  </xacro:macro>
+
+  <link name="base_link"/>
+
+  <xacro:robotiq_hand prefix="l_" parent="base_link" reflect="-1">
+     <origin xyz="0 0 0" rpy="0 0 0"/>
+  </xacro:robotiq_hand>
```

Comment:
Is this on purpose (ie: instantiate xacro macro in the same file that defines it)?